pkg/skaffold/runner/runner_test.go

```diff
@@ -321,3 +322,24 @@ func TestRun(t *testing.T) {
 		})
 	}
 }
+
+func TestGetContextEnvironment(t *testing.T) {
+	os.Setenv("KUBE_CONTEXT", "test-environment-context")
+	defer os.Unsetenv("KUBE_CONTEXT")
+
+	if c := getContext("test-kube-context", ""); c != "test-environment-context" {
+		t.Errorf("expected %q, have %q", "test-environment-context", c)
+	}
+}
+
+func TestParseContextDeployer(t *testing.T) {
+	if c := getContext("test-kube-context", "test-deployer-context"); c != "test-deployer-context" {
+		t.Errorf("expected %q, have %q", "test-deployer-context", c)
```

Comment:
for assertions, you should use the standard skaffold utilities, e.g. `testutil.CheckDeepEqual`.